nova/virt/vmwareapi/special_spawning.py

```diff
@@ -257,7 +257,7 @@ def free_host(self, context):
                             for h, vms in vms_per_host.items()}
 
             # take the one with least memory used
-            host, _ = sorted(mem_per_host.items(), key=lambda (x, y): y)[0]
+            host, _ = sorted(mem_per_host.items(), key=lambda x, y: y)[0]
```

Comment:
This breaks on python2:
```
$ python3
Python 3.5.2 (default, Oct  7 2020, 17:19:02) 
[GCC 5.4.0 20160609] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> mem_per_host = {}
>>> ost, _ = sorted(mem_per_host.items(), key=lambda (x, y): y)[0]
  File "<stdin>", line 1
    ost, _ = sorted(mem_per_host.items(), key=lambda (x, y): y)[0]
                                                     ^
SyntaxError: invalid syntax
>>> 
```
```
$ python2
Python 2.7.12 (default, Oct  5 2020, 13:56:01) 
[GCC 5.4.0 20160609] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> mem_per_host = {}
>>> ost, _ = sorted(mem_per_host.items(), key=lambda (x, y): y)[0]
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
IndexError: list index out of range
>>> 
```
```
$ python3
Python 3.5.2 (default, Oct  7 2020, 17:19:02) 
[GCC 5.4.0 20160609] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> mem_per_host = {}
>>> host, _ = sorted(mem_per_host.items(), key=lambda x, y: y)[0]
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
IndexError: list index out of range
>>> 
```
```
$ python2
Python 2.7.12 (default, Oct  5 2020, 13:56:01) 
[GCC 5.4.0 20160609] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> mem_per_host = {}
>>> host, _ = sorted(mem_per_host.items(), key=lambda x, y: y)[0]
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
IndexError: list index out of range
>>> mem_per_host = {1: 2, 3: 4}
>>> host, _ = sorted(mem_per_host.items(), key=lambda x, y: y)[0]
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: <lambda>() takes exactly 2 arguments (1 given)

Reply:
ah it's not just two args, it's tuple unpacking. yeah, that's gone as well in py3... hm.

Reply:
we could use `attrgetter(1)` I guess

Reply:
I mean `itemgetter(1)`:
```
>>> from operator import itemgetter
>>> host, _ = sorted(mem_per_host.items(), key=itemgetter(1))
```